addons/mail/models/statusbar_time_mixin.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from odoo import api, fields, models
+
+
+class StatusbarTimeMixin(models.AbstractModel):
+    _name = "statusbar.time.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+    _stage_field = "stage_id"
+
+    json_statustime = fields.Char(
```

Comment:
Did you try the `Json` `Field` ? It could allow you to spare the dumps and loads.
(Also I don't see the difference between `statustime` and `time_per_stage`, can't we chose one?)
```suggestion
    time_per_stage = fields.Json(
```
or
```suggestion
    time_per_stage_json = fields.Json(
```

Reply:
Hello @flch-odoo,
It is indeed possible with `fields.Json`, I didn't use it at the time because it says in the code that the field is in beta. Also, for some reason, when using a Json field, `UPDATE` queries are not batched when they have the same content for different ids. This adds a query for each lead in a batch in comparison with a `Char` field. Do you think it is worth the trade-off?

As for `statustime` and `time_per_stage`, I changed it so that there is no reference of `time_per_stage` anymore so that everything is more uniform